src/credentials/CHIPCert.cpp

```diff
@@ -156,13 +156,14 @@ CHIP_ERROR ChipCertificateSet::LoadCert(const uint8_t * chipCert, uint32_t chipC
     err = reader.Next(kTLVType_Structure, ProfileTag(Protocols::OpCredentials::Id.ToTLVProfileId(), kTag_ChipCertificate));
     SuccessOrExit(err);
 
-    err = LoadCert(reader, decodeFlags);
+    err = LoadCert(reader, decodeFlags, chipCert, chipCertLen);
 
 exit:
     return err;
 }
 
-CHIP_ERROR ChipCertificateSet::LoadCert(TLVReader & reader, BitFlags<CertDecodeFlags> decodeFlags)
+CHIP_ERROR ChipCertificateSet::LoadCert(TLVReader & reader, BitFlags<CertDecodeFlags> decodeFlags, const uint8_t * chipCert,
```

Comment:
Please ignore my previous comment...

It will be nice if you summarize the use case for having `mCertificateBegin` `mCertificateLen` parameters. We can then understand what is the best implementation to support that. 

I see some problems with the proposed implementation as not in all cases the pointer and length are available - specifically in cases where the certificate is decoded directly from the TLV buffer...

One of the implementation ideas that I have, where these parameters can be calculated internally:
`cert->mCertificateBegin = reader.GetReadPoint();`
and at the end of decoding:
`cert->mCertificateLen = reader.GetReadPoint() - cert->mCertificateBegin;
`

Reply:
The use case for having those parameters is to store the raw begin address and length of the original CHIP certificate (the one that's being loaded).

This will be useful for serializing the Certificates since we are storing the raw CHIP certificate data. Also, ChipCertificateData is our only valid reference with the certificate data. It holds certificate data in a struct only while the original buffer is 'alive'. Because of that I think it makes sense to reserve some member variables for the original data begin address and length.

Your suggestion seems nice, but it won't work because at that moment, we've already skipped 1 or 2 TLV headers.